src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1032,9 +1032,7 @@ RemainingTestsForCase makeNext(ArrayBuilder<BoundDagTest> remainingTests)
 
             static bool isRuntimeSimilar(TypeSymbol expressionType, TypeSymbol patternType)
             {
-                while (expressionType is ArrayTypeSymbol { ElementType: var e1, IsSZArray: var sz1, Rank: var r1 } &&
-                       patternType is ArrayTypeSymbol { ElementType: var e2, IsSZArray: var sz2, Rank: var r2 } &&
-                       sz1 == sz2 && r1 == r2)
+                while (expressionType is ArrayTypeSymbol { ElementType: var e1, IsSZArray: { } sz1, Rank: { } r1 } && patternType is ArrayTypeSymbol { ElementType: var e2, IsSZArray: var sz2, Rank: var r2 })
```

Comment:
💭 The change from `var sz1` to `{ } sz1` here struck me as odd, but technically it's not incorrect since the type is a non-nullable value type.

Reply:
Unless you actually need to check for null, `var` should be preferred.

---
In reply to: [344383313](https://github.com/dotnet/roslyn/pull/39738#discussion_r344383313) [](ancestors = 344383313)

Reply:
@alrz Action item here is to use `var` instead of `{ }` when the type is a non-nullable value type

Reply:
📝 we should not suggest a fix here because the expression cannot be simplified any further.